src/transformers/models/idefics/processing_idefics.py

```diff
@@ -0,0 +1,340 @@
+        generated_ids = model.generate(**inputs, max_length=100)
+        generated_text = processor.batch_decode(generated_ids, skip_special_tokens=True)[0]
+        ```
+
+        In this example the `prompts` will be converted into:
+
+        ```
+        <s>User:<fake_token_around_image><image><fake_token_around_image>Describe this image.
+        Assistant: An image of two kittens in grass.
+        User:<fake_token_around_image><image><fake_token_around_image>Describe this image.
+        Assistant:'
+        ```
+
+        and the two images will be massaged using [`IdeficsImageProcessor.__call__`] method and placed inside the
+        `pixel_values` dict entry of the return value.
+
+        This example also examplifies that images can be passed as objects or as text urls. It can be seen that the
+        first image is passed as object and the second one as a url.
+
+        In order to help debug prompt generation enable `debug=True` which will show you what's happening.
+
+        """
+
+        # turn non-batched prompts into batched
+        if not any(isinstance(i, list) for i in prompts):
+            prompts = [prompts]
+
+        fake_token = "<fake_token_around_image>"
+        image_token = "<image>"
+
+        def image_tokens(last_was_image):
+            if last_was_image:
+                return image_token + fake_token
+            else:
+                return fake_token + image_token + fake_token
+
+        all_texts = []
+        all_images = []
+        for sample in prompts:
+            # the model was trained on samples starting with <s>
+            full_text = f"{self.tokenizer.bos_token}"
+
+            # an image can either be an image object in the item or the url, everything else is a verbatim prompt text
+            real_images = []
+            last_was_image = False
+            for item in sample:
+                if isinstance(item, str):
+                    item = item.strip(" ")
+                    if is_url(item):
+                        image = self.image_processor.fetch_images(item)
+                        full_text += image_tokens(last_was_image)
+                        real_images.append(image)
+                        last_was_image = True
+                    else:
+                        full_text += item
+                        last_was_image = False
+                else:
+                    # must be an image obj
+                    full_text += image_tokens(last_was_image)
+                    real_images.append(item)
+                    last_was_image = True
+
+            if debug is True:
+                print(f"{full_text=}")
+
+            real_images = self.image_processor(real_images, eval_mode=eval_mode, return_tensors=return_tensors)
+            if len(real_images) > 0:
+                real_images = torch.stack(real_images)
+
+            encoding = self.tokenizer(
+                text=full_text,
+                add_special_tokens=False,
+            )
```

Comment:
Could we allow the typical additional flags for the tokenizer?

In particular, right now, i am interested in `truncation` and `padding` and `max_length`.
This allows me in the demo to generate beyond the max sequence length of idefics (2048 as inherited from llama) by taking the last 1024 (for instance) tokens of the sequence, and generating based on this 1024 conditioning context.

Reply:
oh, sure, it had a gazillion of options - 

1. which ones should we support? other than the 3 you listed? or start there and expand as needed?
2. how should we pass all these in the `processor` - have a tokenizer args dics? or have them flat?

Reply:
it looks like other processors have it flat:

https://github.com/huggingface/transformers/blob/ee4250a35f3bd5e9a4379b4907b3d8f9d5d9523f/src/transformers/models/vilt/processing_vilt.py#L68-L82

Reply:
I pushed the support for the first 3 - just flat 3 new args to `processor`

Reply:
https://github.com/huggingface/transformers/pull/24796/commits/f5029ddfad4ef45046d089a6ed57d7896aac68eb

Reply:
awesome! thank you! yes let's start with these 3 and maybe another use case will appear, but for now, that answers what we need!